src/Npgsql/Counters.cs

```diff
@@ -10,39 +11,39 @@ static class Counters
         /// <summary>
         /// The number of connections per second that are being made to a database server.
         /// </summary>
-        internal static Counter HardConnectsPerSecond;
+        [NotNull] internal static Counter? HardConnectsPerSecond;
```

Comment:
I think this is the kind of place where it's better to keep the variable non-nullable, and force-initialize it to null (with a bang). It gives the same effect as [NotNull] but is a bit simpler.

Reply:
I think it's better to add an `Initialize` method to the counters.

Reply:
Did not understand... Counters.Initialize already exists, and should always be called before any individual counter is accessed (this is why it's safe to say they're non-nullable).